tests/integration/test_rex.py

```diff
@@ -677,3 +677,113 @@ def test_date_functions(c):
             FROM df
             """
         )
+
+
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
+def test_totimestamp(c, gpu):
+    if gpu:
+        import cudf
+
+    df = pd.DataFrame(
+        {
+            "a": np.array([1203073300, 1406073600, 2806073600]),
+        }
+    )
+    if gpu:
+        df = cudf.from_pandas(df)
+    c.create_table("df", df, gpu=gpu)
+
+    df = c.sql(
+        """
+        SELECT to_timestamp(a) AS date FROM df
+    """
+    )
+    expected_df = pd.DataFrame(
+        {
+            "date": [
+                datetime(2008, 2, 15, 11, 1, 40),
+                datetime(2014, 7, 23),
+                datetime(2058, 12, 2, 16, 53, 20),
+            ],
+        }
+    )
+    assert_eq(df, expected_df, check_dtype=False)
+
+    df = c.sql(
+        """
+        SELECT to_timestamp(a, "%d/%m/%Y") AS date FROM df
+    """
+    )
+    expected_df = pd.DataFrame(
+        {
+            "date": [
+                datetime(2008, 2, 15),
+                datetime(2014, 7, 23),
+                datetime(2058, 2, 12),
+            ],
+        }
+    )
+    if not gpu:
+        assert_eq(df, expected_df, check_dtype=False)
```

Comment:
Does this currently fail on GPU? If so interested in what error message we see.

If this does fail currently, would it be possible to split this GPU case off into its own xfailing test so that we can better track it (related to what I'm doing in #867)

Reply:
Yes, I was having trouble getting `dd.to_datetime()` (or `cudf.to_datetime()` or any other datetime functions) to be compatible with cuDF Series, which is fine if we're just using `to_timestamp()` with the default formatting string, but otherwise I couldn't figure out a good way to reformat timestamps without causing errors when `compute()` is called (hence the printed message [here](https://github.com/dask-contrib/dask-sql/blob/df688e7329f0caa8b53fee60130883586622a9e8/dask_sql/physical/rex/core/call.py#L616)).

Right now it's unclear to me whether this would need to be fixed on the Dask side, the cuDF side, or whether there's a Dask-SQL solution I missed. I can open an issue to look further into this if that works?

This is the error message I would get when trying to use a solution with `to_datetime` and `strftime` to format the timestamp:
```
c = <dask_sql.context.Context object at 0x7f3f3ea2b0a0>, gpu = True

    @pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
    def test_totimestamp(c, gpu):
        if gpu:
            import cudf
    
        df = pd.DataFrame(
            {
                "a": np.array([1203073300, 1406073600, 2806073600]),
            }
        )
        if gpu:
            df = cudf.from_pandas(df)
        c.create_table("df", df, gpu=gpu)
    
        df = c.sql(
            """
            SELECT to_timestamp(a) AS date FROM df
        """
        )
        expected_df = pd.DataFrame(
            {
                "date": [
                    datetime(2008, 2, 15, 11, 1, 40),
                    datetime(2014, 7, 23),
                    datetime(2058, 12, 2, 16, 53, 20),
                ],
            }
        )
>       assert_eq(df, expected_df, check_dtype=False)

tests/integration/test_rex.py:710: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/utils.py:13: in assert_eq
    return _assert_eq(*args, **kwargs)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/dataframe/utils.py:545: in assert_eq
    assert_divisions(a, scheduler=scheduler)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/dataframe/utils.py:620: in assert_divisions
    results = get(ddf.dask, ddf.__dask_keys__())
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:557: in get_sync
    return get_async(
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:500: in get_async
    for key, res_info, failed in queue_get(queue).result():
/opt/conda/envs/dask_sql/lib/python3.9/concurrent/futures/_base.py:439: in result
    return self.__get_result()
/opt/conda/envs/dask_sql/lib/python3.9/concurrent/futures/_base.py:391: in __get_result
    raise self._exception
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:542: in submit
    fut.set_result(fn(*args, **kwargs))
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:238: in batch_execute_tasks
    return [execute_task(*a) for a in it]
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:238: in <listcomp>
    return [execute_task(*a) for a in it]
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:229: in execute_task
    result = pack_exception(e, dumps)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/local.py:224: in execute_task
    result = _execute_task(task, data)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/core.py:119: in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/optimization.py:990: in __call__
    return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/core.py:149: in get
    result = _execute_task(task, cache)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/core.py:119: in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/utils.py:71: in apply
    return func(*args, **kwargs)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/dask/dataframe/core.py:6770: in apply_and_enforce
    df = func(*args, **kwargs)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1100: in to_datetime
    result = convert_listlike(argc, format)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:393: in _convert_listlike_datetimes
    return _to_datetime_with_unit(arg, unit, name, tz, errors)
/opt/conda/envs/dask_sql/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:556: in _to_datetime_with_unit
    arg = np.asarray(arg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = 0    1203073300
1    1406073600
2    2806073600
Name: a, dtype: int64
dtype = None

    def __array__(self, dtype=None):
>       raise TypeError(
            "Implicit conversion to a host NumPy array via __array__ is not "
            "allowed, To explicitly construct a GPU matrix, consider using "
            ".to_cupy()\nTo explicitly construct a host matrix, consider "
            "using .to_numpy()."
        )
E       TypeError: Implicit conversion to a host NumPy array via __array__ is not allowed, To explicitly construct a GPU matrix, consider using .to_cupy()
E       To explicitly construct a host matrix, consider using .to_numpy().

/opt/conda/envs/dask_sql/lib/python3.9/site-packages/cudf/core/frame.py:447: TypeError
```

With my current changes, the error is just due to `expected_df` being a truncated version of `df`.

Reply:
Looking a bit deeper it does seem like `dd.to_datetime` doesn't work completely with cudf backed dataframes, 
here's a small reproducer:
```python
import cudf
import dask_cudf

 a = cudf.Series(['2022-01-01', '2022-02-02'])
dser = dask_cudf.from_cudf(a, 1)
res = dd.to_datetime(dser).compute()
```

We could start off with an issue in dask/ dask-cudf and take it from there.